packages/material-renderers/test/renderers/MaterialTableControl.test.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+  The MIT License
+
+  Copyright (c) 2017-2019 EclipseSource Munich
+  https://github.com/eclipsesource/jsonforms
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to deal
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be included in
+  all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+import { JsonSchema7 } from '@jsonforms/core';
+import * as React from 'react';
+
+import { materialCells, materialRenderers } from '../../src';
+import Enzyme, { mount, ReactWrapper } from 'enzyme';
+import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
+import { JsonForms } from '@jsonforms/react';
+import { FormHelperText } from '@mui/material';
+
+Enzyme.configure({ adapter: new Adapter() });
+
+const dataWithEmptyMessage = {
+  nested: [
+    {
+      message: '',
+    },
+  ],
+};
+
+const schemaWithRequired: JsonSchema7 = {
+  type: 'object',
+  properties: {
+    nested: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          message: { type: 'string' },
+          done: { type: 'boolean' },
+        },
+        required: ['message'],
+      },
+    },
+  },
+};
+
+const uischema = {
+  type: 'VerticalLayout',
+  elements: [
+    {
+      type: 'Control',
+      scope: '#/properties/nested',
+    },
+  ],
+};
+
+describe('Material table control', () => {
+  let wrapper: ReactWrapper;
+
+  afterEach(() => wrapper.unmount());
+
+  it('should show required helper for required property with validation mode ValidateAndShow', () => {
+    wrapper = mount(
+      <JsonForms
+        data={dataWithEmptyMessage}
+        schema={schemaWithRequired}
+        uischema={uischema}
+        renderers={materialRenderers}
+        cells={materialCells}
+        validationMode='ValidateAndShow'
+      />
+    );
+    const messageFormHelperText = wrapper.find(FormHelperText).at(0);
+    expect(messageFormHelperText.text()).toBe(
+      "must have required property 'message'"
```

Comment:
@sdirix Thought I'd at least attempt to add relevant unit tests - there didn't seem to be any for `MaterialTableControl`

This fails - not sure if validation just doesn't run for some reason

If you know why let me know and I'll fix it, otherwise I can either skip this test, or remove this file all together.

Testing manually with the examples seems to be working as expected

Reply:
Interestingly, the test works with `message: { type: 'string', minLength: 3 }`

So that's another option, but doesn't really explain why required is not being calculated correctly in this context

Reply:
In the unit test, did you add the required property as an empty string or leave out the property completely?
If it's empty, the required check does not fail. In the ui, the renderer unsets the field to undefined when it is empty. That could be a reason for different results.

I also tested it in the example app and there showing the required error on an array item still worked fine (I locally extended the arrays.ts example for this).

Could you re-add a unit test for this? If it still doesn't run as expected, I can have a look then :)